src/core/server/config/config_service.ts

```diff
@@ -122,24 +125,21 @@ export class ConfigService {
     return config.getFlattenedPaths().filter(path => isPathHandled(path, handledPaths));
   }
 
-  private createConfig<TSchema extends Type<any>, TConfig>(
-    path: ConfigPath,
-    config: Record<string, any>,
-    ConfigClass: ConfigWithSchema<TSchema, TConfig>
-  ) {
-    const namespace = Array.isArray(path) ? path.join('.') : path;
-
-    const configSchema = ConfigClass.schema;
-
-    if (configSchema === undefined || typeof configSchema.validate !== 'function') {
-      throw new Error(
-        `The config class [${
-          ConfigClass.name
-        }] did not contain a static 'schema' field, which is required when creating a config instance`
-      );
+  public async validateAll() {
+    for (const namespace of this.schemas.keys()) {
+      await this.getValidatedConfig(namespace)
+        .pipe(first())
+        .toPromise();
     }
+  }
 
-    const validatedConfig = ConfigClass.schema.validate(
+  private validateConfig(path: ConfigPath, config: Record<string, any>) {
+    const namespace = pathToString(path);
+    const schema = this.schemas.get(namespace);
+    if (!schema) {
+      throw new Error(`No config validator defined for ${namespace}`);
```

Comment:
nit: maybe we should change this error message to: `No config schema defined for ${namespace}`